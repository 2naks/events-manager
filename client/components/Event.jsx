import React, { Component } from 'react';
import moment from 'moment';
import DeleteModal from './DeleteModal.jsx';

export default class Event extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <div className="col-md-4 event">
        <div className="card">
          <div className="card-block">
            <h5 className="card-title"><b>{this.props.eventDetails.title}</b></h5>
            <p className="card-text h6"> {this.props.eventDetails.description}</p>
            <p className="card-text h6"><b>Venue</b> {this.props.eventDetails.venue}</p>
            <p className="card-text h6"><b>Date</b>: {moment(this.props.eventDetails.date).format('DD MMMM YYYY')}</p>
            <p className="card-text h6"><b>Time</b>: {this.props.eventDetails.time}</p>
            <p className="text-center"><a href={`/events/${this.props.eventDetails.id}`} className="btn btn-sm btn-primary"><i className="fa fa-info-circle fa-lg" /> View Event</a>
              <a href={`./modifyevent/${this.props.eventDetails.id}`} className="btn btn-sm btn-success"><i className="fa fa-edit fa-lg" /> Modify</a>
              <a href="#" className="btn btn-sm btn-danger" data-toggle="modal" data-target={`#${this.props.eventDetails.id}`}><i className="fa fa-trash-o fa-lg" /> Delete</a></p>
          </div>
        </div>
        <DeleteModal item="event" objectId={this.props.eventDetails.id} />
      </div>
    );
  }
}
