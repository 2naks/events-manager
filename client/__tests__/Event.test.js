import React, { Components } from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import Adapter from 'enzyme-adapter-react-16';
import Event  from '../components/Event.jsx';

Enzyme.configure({ adapter: new Adapter() });

describe('Event Component', () => {

  const props = {
      eventDetails:{
        id: '1',
        title: 'Seminar',
        description: "",
        location: "",
        time:"5:00",
        date:Date.now(),
        image: "",
        Center:[]
      }
    };

  describe('Event  should render self', () => {
    it('it should render for dumb component', () => {
      const wrapper = shallow(<Event {...props} />);
      expect(wrapper).toMatchSnapshot();
    });
  });

});
