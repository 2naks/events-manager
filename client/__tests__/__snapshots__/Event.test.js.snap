// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Component Event  should render self it should render for dumb component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Event
    eventDetails={
      Object {
        "Center": Array [],
        "date": 1515401550943,
        "description": "",
        "id": "1",
        "image": "",
        "location": "",
        "time": "5:00",
        "title": "Seminar",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="card"
        >
          <div
            className="card-block"
          >
            <h5
              className="card-title"
            >
              <b>
                Seminar
              </b>
            </h5>
            <p
              className="card-text h6"
            >
               
              
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Venue
              </b>
               
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Date
              </b>
              : 
              08 January 2018
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Time
              </b>
              : 
              5:00
            </p>
            <p
              className="text-center"
            >
              <a
                className="btn btn-sm btn-primary"
                href="/events/1"
              >
                <i
                  className="fa fa-info-circle fa-lg"
                />
                 View Event
              </a>
              <a
                className="btn btn-sm btn-success"
                href="./modifyevent/1"
              >
                <i
                  className="fa fa-edit fa-lg"
                />
                 Modify
              </a>
              <a
                className="btn btn-sm btn-danger"
                data-target="#1"
                data-toggle="modal"
                href="#"
              >
                <i
                  className="fa fa-trash-o fa-lg"
                />
                 Delete
              </a>
            </p>
          </div>
        </div>,
        <DeleteModal
          objectId="1"
        />,
      ],
      "className": "col-md-4 event",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="card-block"
          >
            <h5
              className="card-title"
            >
              <b>
                Seminar
              </b>
            </h5>
            <p
              className="card-text h6"
            >
               
              
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Venue
              </b>
               
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Date
              </b>
              : 
              08 January 2018
            </p>
            <p
              className="card-text h6"
            >
              <b>
                Time
              </b>
              : 
              5:00
            </p>
            <p
              className="text-center"
            >
              <a
                className="btn btn-sm btn-primary"
                href="/events/1"
              >
                <i
                  className="fa fa-info-circle fa-lg"
                />
                 View Event
              </a>
              <a
                className="btn btn-sm btn-success"
                href="./modifyevent/1"
              >
                <i
                  className="fa fa-edit fa-lg"
                />
                 Modify
              </a>
              <a
                className="btn btn-sm btn-danger"
                data-target="#1"
                data-toggle="modal"
                href="#"
              >
                <i
                  className="fa fa-trash-o fa-lg"
                />
                 Delete
              </a>
            </p>
          </div>,
          "className": "card",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h5
                className="card-title"
              >
                <b>
                  Seminar
                </b>
              </h5>,
              <p
                className="card-text h6"
              >
                 
                
              </p>,
              <p
                className="card-text h6"
              >
                <b>
                  Venue
                </b>
                 
              </p>,
              <p
                className="card-text h6"
              >
                <b>
                  Date
                </b>
                : 
                08 January 2018
              </p>,
              <p
                className="card-text h6"
              >
                <b>
                  Time
                </b>
                : 
                5:00
              </p>,
              <p
                className="text-center"
              >
                <a
                  className="btn btn-sm btn-primary"
                  href="/events/1"
                >
                  <i
                    className="fa fa-info-circle fa-lg"
                  />
                   View Event
                </a>
                <a
                  className="btn btn-sm btn-success"
                  href="./modifyevent/1"
                >
                  <i
                    className="fa fa-edit fa-lg"
                  />
                   Modify
                </a>
                <a
                  className="btn btn-sm btn-danger"
                  data-target="#1"
                  data-toggle="modal"
                  href="#"
                >
                  <i
                    className="fa fa-trash-o fa-lg"
                  />
                   Delete
                </a>
              </p>,
            ],
            "className": "card-block",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <b>
                  Seminar
                </b>,
                "className": "card-title",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Seminar",
                },
                "ref": null,
                "rendered": "Seminar",
                "type": "b",
              },
              "type": "h5",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  " ",
                  "",
                ],
                "className": "card-text h6",
              },
              "ref": null,
              "rendered": Array [
                " ",
                "",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Venue
                  </b>,
                  " ",
                  undefined,
                ],
                "className": "card-text h6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Venue",
                  },
                  "ref": null,
                  "rendered": "Venue",
                  "type": "b",
                },
                " ",
                undefined,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Date
                  </b>,
                  ": ",
                  "08 January 2018",
                ],
                "className": "card-text h6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Date",
                  },
                  "ref": null,
                  "rendered": "Date",
                  "type": "b",
                },
                ": ",
                "08 January 2018",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <b>
                    Time
                  </b>,
                  ": ",
                  "5:00",
                ],
                "className": "card-text h6",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Time",
                  },
                  "ref": null,
                  "rendered": "Time",
                  "type": "b",
                },
                ": ",
                "5:00",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <a
                    className="btn btn-sm btn-primary"
                    href="/events/1"
                  >
                    <i
                      className="fa fa-info-circle fa-lg"
                    />
                     View Event
                  </a>,
                  <a
                    className="btn btn-sm btn-success"
                    href="./modifyevent/1"
                  >
                    <i
                      className="fa fa-edit fa-lg"
                    />
                     Modify
                  </a>,
                  <a
                    className="btn btn-sm btn-danger"
                    data-target="#1"
                    data-toggle="modal"
                    href="#"
                  >
                    <i
                      className="fa fa-trash-o fa-lg"
                    />
                     Delete
                  </a>,
                ],
                "className": "text-center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fa fa-info-circle fa-lg"
                      />,
                      " View Event",
                    ],
                    "className": "btn btn-sm btn-primary",
                    "href": "/events/1",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-info-circle fa-lg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " View Event",
                  ],
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fa fa-edit fa-lg"
                      />,
                      " Modify",
                    ],
                    "className": "btn btn-sm btn-success",
                    "href": "./modifyevent/1",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-edit fa-lg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " Modify",
                  ],
                  "type": "a",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <i
                        className="fa fa-trash-o fa-lg"
                      />,
                      " Delete",
                    ],
                    "className": "btn btn-sm btn-danger",
                    "data-target": "#1",
                    "data-toggle": "modal",
                    "href": "#",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fa fa-trash-o fa-lg",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                    " Delete",
                  ],
                  "type": "a",
                },
              ],
              "type": "p",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "objectId": "1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="card"
          >
            <div
              className="card-block"
            >
              <h5
                className="card-title"
              >
                <b>
                  Seminar
                </b>
              </h5>
              <p
                className="card-text h6"
              >
                 
                
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Venue
                </b>
                 
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Date
                </b>
                : 
                08 January 2018
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Time
                </b>
                : 
                5:00
              </p>
              <p
                className="text-center"
              >
                <a
                  className="btn btn-sm btn-primary"
                  href="/events/1"
                >
                  <i
                    className="fa fa-info-circle fa-lg"
                  />
                   View Event
                </a>
                <a
                  className="btn btn-sm btn-success"
                  href="./modifyevent/1"
                >
                  <i
                    className="fa fa-edit fa-lg"
                  />
                   Modify
                </a>
                <a
                  className="btn btn-sm btn-danger"
                  data-target="#1"
                  data-toggle="modal"
                  href="#"
                >
                  <i
                    className="fa fa-trash-o fa-lg"
                  />
                   Delete
                </a>
              </p>
            </div>
          </div>,
          <DeleteModal
            objectId="1"
          />,
        ],
        "className": "col-md-4 event",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="card-block"
            >
              <h5
                className="card-title"
              >
                <b>
                  Seminar
                </b>
              </h5>
              <p
                className="card-text h6"
              >
                 
                
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Venue
                </b>
                 
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Date
                </b>
                : 
                08 January 2018
              </p>
              <p
                className="card-text h6"
              >
                <b>
                  Time
                </b>
                : 
                5:00
              </p>
              <p
                className="text-center"
              >
                <a
                  className="btn btn-sm btn-primary"
                  href="/events/1"
                >
                  <i
                    className="fa fa-info-circle fa-lg"
                  />
                   View Event
                </a>
                <a
                  className="btn btn-sm btn-success"
                  href="./modifyevent/1"
                >
                  <i
                    className="fa fa-edit fa-lg"
                  />
                   Modify
                </a>
                <a
                  className="btn btn-sm btn-danger"
                  data-target="#1"
                  data-toggle="modal"
                  href="#"
                >
                  <i
                    className="fa fa-trash-o fa-lg"
                  />
                   Delete
                </a>
              </p>
            </div>,
            "className": "card",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h5
                  className="card-title"
                >
                  <b>
                    Seminar
                  </b>
                </h5>,
                <p
                  className="card-text h6"
                >
                   
                  
                </p>,
                <p
                  className="card-text h6"
                >
                  <b>
                    Venue
                  </b>
                   
                </p>,
                <p
                  className="card-text h6"
                >
                  <b>
                    Date
                  </b>
                  : 
                  08 January 2018
                </p>,
                <p
                  className="card-text h6"
                >
                  <b>
                    Time
                  </b>
                  : 
                  5:00
                </p>,
                <p
                  className="text-center"
                >
                  <a
                    className="btn btn-sm btn-primary"
                    href="/events/1"
                  >
                    <i
                      className="fa fa-info-circle fa-lg"
                    />
                     View Event
                  </a>
                  <a
                    className="btn btn-sm btn-success"
                    href="./modifyevent/1"
                  >
                    <i
                      className="fa fa-edit fa-lg"
                    />
                     Modify
                  </a>
                  <a
                    className="btn btn-sm btn-danger"
                    data-target="#1"
                    data-toggle="modal"
                    href="#"
                  >
                    <i
                      className="fa fa-trash-o fa-lg"
                    />
                     Delete
                  </a>
                </p>,
              ],
              "className": "card-block",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <b>
                    Seminar
                  </b>,
                  "className": "card-title",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Seminar",
                  },
                  "ref": null,
                  "rendered": "Seminar",
                  "type": "b",
                },
                "type": "h5",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    " ",
                    "",
                  ],
                  "className": "card-text h6",
                },
                "ref": null,
                "rendered": Array [
                  " ",
                  "",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b>
                      Venue
                    </b>,
                    " ",
                    undefined,
                  ],
                  "className": "card-text h6",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Venue",
                    },
                    "ref": null,
                    "rendered": "Venue",
                    "type": "b",
                  },
                  " ",
                  undefined,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b>
                      Date
                    </b>,
                    ": ",
                    "08 January 2018",
                  ],
                  "className": "card-text h6",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Date",
                    },
                    "ref": null,
                    "rendered": "Date",
                    "type": "b",
                  },
                  ": ",
                  "08 January 2018",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <b>
                      Time
                    </b>,
                    ": ",
                    "5:00",
                  ],
                  "className": "card-text h6",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Time",
                    },
                    "ref": null,
                    "rendered": "Time",
                    "type": "b",
                  },
                  ": ",
                  "5:00",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <a
                      className="btn btn-sm btn-primary"
                      href="/events/1"
                    >
                      <i
                        className="fa fa-info-circle fa-lg"
                      />
                       View Event
                    </a>,
                    <a
                      className="btn btn-sm btn-success"
                      href="./modifyevent/1"
                    >
                      <i
                        className="fa fa-edit fa-lg"
                      />
                       Modify
                    </a>,
                    <a
                      className="btn btn-sm btn-danger"
                      data-target="#1"
                      data-toggle="modal"
                      href="#"
                    >
                      <i
                        className="fa fa-trash-o fa-lg"
                      />
                       Delete
                    </a>,
                  ],
                  "className": "text-center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <i
                          className="fa fa-info-circle fa-lg"
                        />,
                        " View Event",
                      ],
                      "className": "btn btn-sm btn-primary",
                      "href": "/events/1",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "fa fa-info-circle fa-lg",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                      " View Event",
                    ],
                    "type": "a",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <i
                          className="fa fa-edit fa-lg"
                        />,
                        " Modify",
                      ],
                      "className": "btn btn-sm btn-success",
                      "href": "./modifyevent/1",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "fa fa-edit fa-lg",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                      " Modify",
                    ],
                    "type": "a",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        <i
                          className="fa fa-trash-o fa-lg"
                        />,
                        " Delete",
                      ],
                      "className": "btn btn-sm btn-danger",
                      "data-target": "#1",
                      "data-toggle": "modal",
                      "href": "#",
                    },
                    "ref": null,
                    "rendered": Array [
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "className": "fa fa-trash-o fa-lg",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": "i",
                      },
                      " Delete",
                    ],
                    "type": "a",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "objectId": "1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
